index.ts

```diff
@@ -60,6 +60,7 @@ io.on("connection", socket => {
       const remaining = rooms[roomID].length - 1;
       if (remaining > 0) {
         socket.broadcast.to(roomID).emit("room-user-count", remaining);
+        socket.broadcast.to(roomID).emit("client-disconnected", socket.id);
```

Comment:
It's very suboptimal to be sending two messages. One can get lost and things can get out of sync.

Suggestion, send the list of all the ids within the "room-user-count" event. And leave it up to the client to do the diff to figure out who dropped out and do the necessary action based on this.

Reply:
updated